### PR TITLE
Add pre-commit and lint

### PR DESCRIPTION
**Pull request by jianyunchao**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation
add pre-commit，implement incremental file scanning;
remove runtime/references for cpplint.

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

**Review by SamitHuang — APPROVED** [review]

**Comment by SamitHuang** [comment]
Please add contributing guideline  CONTRIBUTING.md for the changes (pre-commit).

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[flake8]
+max-line-length = 120
+extend-exclude =
+    docs/*,
+    examples/*,
+    setup.py
+extend-ignore = E203
+per-file-ignores =
+    */__init__.py:F401,F403
+    mindaudio/data/__init__.py:F403,F405
```
Comment (HaoyangLee) [inline .flake8:10]: `mindaudio` typo?

File: pylintrc [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+# Disable the message, report, category or checker with the given id(s). You
+# can either give multiple identifiers separated by comma (,) or put this
+# option multiple times (only on the command line, not in the configuration
+# file where it should appear only once).You can also use "--disable=all" to
+# disable everything first and then reenable specific checks. For example, if
+# you want to run only the similarities checker, you can use "--disable=all
+# --enable=similarities". If you want to run only the classes checker, but have
+# no Warning level messages displayed, use"--disable=all --enable=classes
+# --disable=W"
+disable=design,similarities,no-self-use,attribute-defined-outside-init,locally-disabled,star-args,pointless-except,bad-option-value,global-statement,fixme,suppressed-message,useless-suppression,locally-enabled,no-member,no-name-in-module,import-error,unsubscriptable-object,unbalanced-tuple-unpacking,undefined-variable,not-context-manager,wrong-import-position,wrong-import-order
```
Comment (SamitHuang) [inline pylintrc:41]: To ensure the formatted code can run correctly, we also need to disable:

ungrouped-imports, unused-argument

File: .flake8 (hunk 1 above)
Comment (jianyunchao) [reply .flake8]: resolved

File: pylintrc (hunk 2 above)
Comment (jianyunchao) [reply pylintrc]: remove pylint

**Pull request closed without merging**